Proofread parts of the container guide, add tips & tricks section for rootless containers (#72)
SUSE/doc-unversioned: container-guide/adoc/containers-podman-overview.adoc

You can then grant a specific user write access by adding them to the created group.

[[sec-podman-rootless-containers-tricks]]

==== Tips and tricks for rootless containers

> {podman} remaps user IDs with rootless containers. For example {podman} will remap your current user
> to the default user in the container:

Reviewer: ```suggestion
{podman} remaps user IDs with rootless containers. For example, the following command remaps the current user to the default user in the container:
```

  File: /share/
  Size: 318             Blocks: 0          IO Block: 4096   directory
Device: 2ch/44d Inode: 3506170     Links: 1
Access: (0755/drwxr-xr-x)  Uid: (    0/    root)   Gid: (    0/    root)
Access: 2023-05-03 12:52:18.636392618 +0000
Modify: 2023-05-03 12:52:17.178380923 +0000
Change: 2023-05-03 12:52:17.178380923 +0000
 Birth: 2023-05-03 12:52:15.852370288 +0000

 ❯ podman run --userns=keep-id --rm -it -v $(pwd):/share/ registry.suse.com/bci/bci-base stat /share/
  File: /share/
  Size: 318             Blocks: 0          IO Block: 4096   directory
Device: 2ch/44d Inode: 3506170     Links: 1
Access: (0755/drwxr-xr-x)  Uid: ( 1000/     user)   Gid: ( 1000/     users)
Access: 2023-05-03 12:52:18.636392618 +0000
Modify: 2023-05-03 12:52:17.178380923 +0000
Change: 2023-05-03 12:52:17.178380923 +0000
 Birth: 2023-05-03 12:52:15.852370288 +0000
....

{podman} stores your containers data in the storage graph root (by default
`~/.local/share/containers/storage`). It is possible that there are files inside the graph root not
owned by your user due to the way {podman} maps non-root users in containers. Files owned by such a

> remapped non-root user are often not readable by your current user.

Author: I have reworded this paragraph.

In case you have to read or modify any files in the graph root, you can enter a shell with
privileges to do so as follows:

....
 ❯ podman unshare bash

 ❯ id
uid=0(root) gid=0(root) groups=0(root),65534(nobody)
....

Note that `podman unshare` performs the same user remapping as `podman run` does when launching a
rootless container. You cannot gain elevated privileges via `podman unshare`.

==== Caveats of rootless containers

> Rootless containers currently suffer a substantial performance penalty for network transfers. This
> penalty is caused by the inability of unprivileged users to configure network namespaces on
> Linux. Therefore {podman} relies on a userspace network implementation called `slirp4netns`. This
> userspace networking has to emulate the full TCP-IP stack and can cause a heavy performance
> degradation for workloads relying on high network transfer rates.
> 
> On Linux unprivileged users cannot open ports below port number 1024. This limitation also applies
> to {podman} and therefore rootless containers can by default not expose ports below port number
> 1024. You can lift this limitation via:

Reviewer: ```suggestion
Because unprivileged users cannot configure network namespaces on
Linux, {podman} relies on a userspace network implementation called `slirp4netns`. It emulates the full TCP-IP stack and can cause a heavy performance degradation for workloads relying on high network transfer rates. This means that rootless containers suffer from slow network transfers.

On Linux, unprivileged users cannot open ports below port number 1024. This limitation also applies
to {podman}, so by default, rootless containers cannot expose ports below port number
1024. You can remove this limitation using the following command:
```

....
sysctl net.ipv4.ip_unprivileged_port_start=0
....
or make it persistent via:
....
sysctl -w net.ipv4.ip_unprivileged_port_start=0
....

Note that this allows *all* unprivileged applications to bind to ports below 1024.

[[sec-podman-docker]]
==== {podman}-docker

Since {podman} is compatible with {deng}, it features the same command line interface. You can also
install the package `podman-docker`, which will allow you to use an emulated docker CLI using
podman, for example, the `docker pull` command will actually execute `podman pull`, which fetches a

> container image from a registry, or `docker build` will execute `podman build`.

Author: I've taken your suggestion and modified it a bit.

{podman} also features a {deng} compatible socket that can be launched via:
....
 ❯ sudo systemctl start podman.socket
....

The {podman} socket can be used by applications designed to communicate with {deng} to launch
containers transparently via {podman} instead. The {podman} socket can be used for instance to
launch containers via `docker compose` without having to run {deng}.

[[sec-podman-get]]
=== Obtaining container images

[[sec-podman-registries]]
==== Configuring container registries

One of the advantages of {podman} over {deng} is that {podman} can be configured to search multiple container registries. To make {podman} search the {suseregistry} first and use {dhub} as a fallback, add the following configuration to the `/etc/containers/registries.conf` file:

....
unqualified-search-registries = ["registry.suse.com", "docker.io"]
....

> Note that this step is unnecessary on {slea} or openSUSE. On both distributions {suseregistry}
> and `registry.opensuse.org` come before {dhub}.

Reviewer: ```suggestion
Note that this step can be skipped on {slea} or openSUSE. On both distributions, {suseregistry}
and `registry.opensuse.org` have priority over {dhub}.
```
